packages/core/fs/src/NodeFS.js

```diff
@@ -222,11 +224,43 @@ try {
   //
 }
 
+let useOsTmpDir;
+function shouldUseOsTmpDir(filePath) {
+  if (useOsTmpDir != null) {
+    return useOsTmpDir;
+  }
+  try {
+    const tmpDir = tmpdir();
+    nativeFS.accessSync(
+      tmpDir,
+      nativeFS.constants.R_OK | nativeFS.constants.W_OK,
+    );
+    const tmpDirStats = nativeFS.statSync(tmpDir);
+    const filePathStats = nativeFS.statSync(filePath);
+    // Check the tmpdir is on the same partition as the target directory.
+    // This is required to allow renaming to occur.
```

Comment:
Technically it's required to ensure renaming is an atomic operation - you can still rename between filesystems, it'll just be a copy/delete which isn't atomic.